internal/peer/node/start.go

```diff
@@ -495,14 +495,17 @@ func serve(args []string) error {
 		CCID:            scc.ChaincodeID(lifecycle.LifecycleNamespace),
 		HandlerRegistry: chaincodeHandlerRegistry,
 	}
-	var client *docker.Client
-	if coreConfig.VMDockerTLSEnabled {
-		client, err = docker.NewTLSClient(coreConfig.VMEndpoint, coreConfig.DockerCert, coreConfig.DockerKey, coreConfig.DockerCA)
-	} else {
-		client, err = docker.NewClient(coreConfig.VMEndpoint)
+
+	if coreConfig.VMEndpoint == "" && len(coreConfig.ExternalBuilders) == 0 {
+		logger.Panic("VMEndpoint not set and no ExternalBuilders defined")
 	}
-	if err != nil {
-		logger.Panicf("cannot create docker client: %s", err)
+
+	var client *docker.Client
+	if coreConfig.VMEndpoint != "" {
```

Comment:
shouldn't we also move the construct of `dockerVM` into this branch? and at fallback point, if `router.DockerVM == nil` we would return early and skip package retrieval.